### DISCUSSION TITLE
Any chance of moving from `boolean | 'always'` to unions for `refetchOnX`

### DISCUSSION THREAD

**Discussion by lessp** [TanStack/query#3443] [General]
Hey folks, thanks for a great library!

With a major version bump approaching, is there any chance you might consider clarifying the `refetchOnX` API to using e.g. `'always' | 'ifStale' | 'never'` (or similar) instead of the rather avant-garde `boolean | 'always'`?

Would be more than happy to take this on. 🙂

**Comment by TkDodo** [comment]
haven't given it much thought, but I'd rather say no :). The `always` is already rather an edge case - you likely don't want to refetch "fresh" data. If that's what you want, invalidating it before is likely better anyways. So most of the time, you only want the two states, which are represented by the boolean.
